### PR TITLE
Transferred all file thread logs and file thread function dependencies to their equivalent ocp diag reprsentation

### PR DESCRIPTION
**Pull request by dhawkes**
Internal reference: 275900374

Tested:
dthawkes@dthawkes:~/ocp-diag-sat$ ./bazel-bin/src/sat_bin -f /tmp/tpm.1234 -m 1
{"schemaVersion":{"major":2,"minor":0},"sequenceNumber":0,"timestamp":"2023-04-26T01:29:56Z"}
{"testRunArtifact":{"testRunStart":{"name":"Stress App Test","version":"1.0.0","commandLine":"./bazel-bin/src/sat_bin -f /tmp/tpm.1234 -m 1","parameters":{"f":"/tmp/tpm.1234","m":"1"},"dutInfo":{"dutInfoId":"holder","name":"place","metadata":{},"platformInfos":[],"hardwareInfos":[],"softwareInfos":[]},"metadata":{}}},"sequenceNumber":1,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"testStepStart":{"name":"Setup and Check Environment"},"testStepId":"0"},"sequenceNumber":2,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"CPU has clflush and has sse2."},"testStepId":"0"},"sequenceNumber":3,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"CPU Core Count","unit":"cores","hardwareInfoId":"","validators":[],"metadata":{},"value":64},"testStepId":"0"},"sequenceNumber":4,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Node Count","unit":"nodes","hardwareInfoId":"","validators":[],"metadata":{},"value":1},"testStepId":"0"},"sequenceNumber":5,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Total memory: 515947 MB, Free memory: 496848 MB, Hugepage memory: 0 MB. Targetting 489958 MB (94%) for testing."},"testStepId":"0"},"sequenceNumber":6,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"INFO","message":"Preferring plain malloc memory allocation."},"testStepId":"0"},"sequenceNumber":7,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"INFO","message":"Using mmap allocation at 0x7efb9ea1e000."},"testStepId":"0"},"sequenceNumber":8,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Memory to Test","unit":"MB","hardwareInfoId":"","validators":[],"metadata":{},"value":489958},"testStepId":"0"},"sequenceNumber":9,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Test Run Time","unit":"s","hardwareInfoId":"","validators":[],"metadata":{},"value":20},"testStepId":"0"},"sequenceNumber":10,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"testStepEnd":{"status":"COMPLETE"},"testStepId":"0"},"sequenceNumber":11,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"testStepStart":{"name":"Setup and Fill Memory Pages"},"testStepId":"1"},"sequenceNumber":12,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Total Memory Page Count","unit":"pages","hardwareInfoId":"","validators":[],"metadata":{},"value":489958},"testStepId":"1"},"sequenceNumber":13,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Required Thread Memory Page Count","unit":"pages","hardwareInfoId":"","validators":[],"metadata":{},"value":2},"testStepId":"1"},"sequenceNumber":14,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"measurement":{"name":"Free Memory Page Count","unit":"pages","hardwareInfoId":"","validators":[{"name":"","type":"GREATER_THAN_OR_EQUAL","value":2},{"name":"","type":"LESS_THAN_OR_EQUAL","value":244979}],"metadata":{},"value":195982},"testStepId":"1"},"sequenceNumber":15,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Allocating memory pages, Total Pages: 489958, Free Pages: 195982"},"testStepId":"1"},"sequenceNumber":16,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill threads: 8 threads, 489958 pages"},"testStepId":"1"},"sequenceNumber":17,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 0 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":18,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 1 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":19,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 2 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":20,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 3 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":21,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 4 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":22,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 5 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":23,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 6 to fill 61244 pages"},"testStepId":"1"},"sequenceNumber":24,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Starting memory page fill Thread 7 to fill 61250 pages"},"testStepId":"1"},"sequenceNumber":25,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #0: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":26,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #1: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":27,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #3: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":28,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #7: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":29,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #5: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":30,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #2: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":31,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #6: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":32,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #4: Starting memory page fill thread"},"testStepId":"1"},"sequenceNumber":33,"timestamp":"2023-04-26T01:29:56Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #5: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":34,"timestamp":"2023-04-26T01:30:09Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #1: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":35,"timestamp":"2023-04-26T01:30:09Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #0: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":36,"timestamp":"2023-04-26T01:30:10Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #4: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":37,"timestamp":"2023-04-26T01:30:10Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #2: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":38,"timestamp":"2023-04-26T01:30:10Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #7: Completed. Status: Success. Filled 61250 pages."},"testStepId":"1"},"sequenceNumber":39,"timestamp":"2023-04-26T01:30:10Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #3: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":40,"timestamp":"2023-04-26T01:30:10Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Page Fill Thread #6: Completed. Status: Success. Filled 61244 pages."},"testStepId":"1"},"sequenceNumber":41,"timestamp":"2023-04-26T01:30:11Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Done filling memory pages. Starting to allocate pages."},"testStepId":"1"},"sequenceNumber":42,"timestamp":"2023-04-26T01:30:11Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Done allocating pages."},"testStepId":"1"},"sequenceNumber":43,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Region 0 corresponds to 489958"},"testStepId":"1"},"sequenceNumber":44,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Region mask: 0x1"},"testStepId":"1"},"sequenceNumber":45,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"testStepEnd":{"status":"COMPLETE"},"testStepId":"1"},"sequenceNumber":46,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"testStepStart":{"name":"Run Memory Copy Threads"},"testStepId":"2"},"sequenceNumber":47,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"testStepStart":{"name":"Run File IO Threads"},"testStepId":"3"},"sequenceNumber":48,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"File IO Thread #1: Starting file thread using file: /tmp/tpm.1234"},"testStepId":"3"},"sequenceNumber":49,"timestamp":"2023-04-26T01:30:14Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Copy Thread #0: Starting memory copy thread. CPU: 1, Mem: ffffffff, Warming: No, Has Vector: Yes"},"testStepId":"2"},"sequenceNumber":50,"timestamp":"2023-04-26T01:30:14Z"}
2023/04/26-01:30:24(UTC) Log: Seconds remaining: 10
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"Memory Copy Thread #0: Status: Success, 192827 pages copied."},"testStepId":"2"},"sequenceNumber":51,"timestamp":"2023-04-26T01:30:34Z"}
{"testStepArtifact":{"log":{"severity":"DEBUG","message":"File IO Thread #1: Completed 1: file thread status 0, 6632 pages copied"},"testStepId":"3"},"sequenceNumber":52,"timestamp":"2023-04-26T01:30:34Z"}
{"testStepArtifact":{"testStepStart":{"name":"Run Post-Test Memory Check Threads"},"testStepId":"4"},"sequenceNumber":53,"timestamp":"2023-04-26T01:30:34Z"}
{"testStepArtifact":{"testStepEnd":{"status":"COMPLETE"},"testStepId":"4"},"sequenceNumber":54,"timestamp":"2023-04-26T01:30:37Z"}
2023/04/26-01:30:37(UTC) Stats: Found 0 hardware incidents
2023/04/26-01:30:37(UTC) Stats: Completed: 398918.00M in 20.02s 19922.85MB/s, with 0 hardware incidents, 0 errors
2023/04/26-01:30:37(UTC) Stats: Memory Copy: 385654.00M at 19281.79MB/s
2023/04/26-01:30:37(UTC) Stats: File Copy: 13264.00M at 662.43MB/s
2023/04/26-01:30:37(UTC) Stats: Net Copy: 0.00M at 0.00MB/s
2023/04/26-01:30:37(UTC) Stats: Data Check: 0.00M at 0.00MB/s
2023/04/26-01:30:37(UTC) Stats: Invert Data: 0.00M at 0.00MB/s
2023/04/26-01:30:37(UTC) Stats: Disk: 0.00M at 0.00MB/s
{"testStepArtifact":{"testStepEnd":{"status":"COMPLETE"},"testStepId":"2"},"sequenceNumber":55,"timestamp":"2023-04-26T01:30:37Z"}
{"testStepArtifact":{"testStepEnd":{"status":"COMPLETE"},"testStepId":"3"},"sequenceNumber":56,"timestamp":"2023-04-26T01:30:37Z"}
2023/04/26-01:30:37(UTC)
2023/04/26-01:30:37(UTC) Status: PASS - please verify no corrected errors
2023/04/26-01:30:37(UTC)
{"testRunArtifact":{"testRunEnd":{"status":"COMPLETE","result":"PASS"}},"sequenceNumber":57,"timestamp":"2023-04-26T01:30:39Z"}

### CODE REVIEW

**Comment by dhawkes** [comment]
@kimmidi cc

**Comment by dhawkes** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/10" target="_blank">Graphite</a> 
    * **PR #12** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/12" target="_blank">Graphite</a> 
      * **PR #11** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/11" target="_blank">Graphite</a> 
        * **PR #13** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/13" target="_blank">Graphite</a> 
          * **PR #14** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/14" target="_blank">Graphite</a> 
            * **PR #15** <a href="https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15?utm_source=stack-comment).

File: src/error_diag.cc [hunk 1]
```diff
@@ -172,7 +172,7 @@ int ErrorDiag::AddMiscompareError(string dimm_string, uint64 addr, int count) {
   error->severity_ = SAT_ERROR_FATAL;
   error->addr_ = addr;
   dimm_device->AddErrorInstance(error);
-  os_->ErrorReport(dimm_string.c_str(), "miscompare", count);
+  // os_->ErrorReport(dimm_string.c_str(), "miscompare", count);
```
Comment (jcgao1) [inline src/error_diag.cc:175]: Why don't you simply remove this? Same below

Comment (jcgao1) [reply src/error_diag.cc:175]: And you have removed `ErrorReport` function.

**Review by jcgao1 — APPROVED** [review]

**Comment by dhawkes** [comment]
@dhawkes started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15).

**Comment by dhawkes** [comment]
@dhawkes started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15).

**Comment by dhawkes** [comment]
@dhawkes started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15).

**Comment by dhawkes** [comment]
@dhawkes merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opencomputeproject/ocp-diag-sat/15).

**Pull request merged**